Add Support to Export to DBML

jpraetorius opened the pull request:
- Create basic project info from what the datacontract has
- Map all the Models into tables, _not_ taking into regard nested fields (as there is no way to express them anyway)
- Also create references, when they are given for a field, so the connections between tables become visible

jpraetorius commented:
Fixes #135

simonharrer commented:
A few comments:
- Hm, tests seem to be failing.
- Add in README and CHANGELOG
- Add a comment in the genereated DBML to say it was generated by the Data Contract CLI and its version. Maybe even add infos like the ID of the data contract.

simonharrer commented:
I'm not sure what types we should present there. I think it might be valuable using the logical types of the datacontract.yaml, but it could also be nice with the type mapping of a particular server technology (snowflake, postgres, etc.). Regardless, what types are used should be stated in the comments at the start of the generated file.

jpraetorius commented:
* it actually were `ruff` checks I failed to catch before :/ – ✅
* Adapted `CHANGELOG` and `README` ✅
* added some information that this is a generated file and who created it when for what datacontract ✅
* added the possibility to select a `--server` so we can convert the abstract types to concrete ones (this is currently reusing the existing SQL Export functionality, which is a bit of a wild import but seemed more reasonable than implementing this again) ✅

simonharrer commented:
Love it. Thanks for the contribution!
